File: selenium/test/java/com/thoughtworks/selenium/corebased/TestClick.java
package com.thoughtworks.selenium.corebased;

import com.thoughtworks.selenium.SeleneseTestNgHelper;
import org.testng.annotations.Test;

public class TestClick extends SeleneseTestNgHelper {
	@Test public void testClick() throws Exception {
		selenium.open("../tests/html/test_click_page1.html");
		verifyEquals(selenium.getText("link"), "Click here for next page");
		selenium.click("link");
		selenium.waitForPageToLoad("30000");
		verifyEquals(selenium.getTitle(), "Click Page Target");
		selenium.click("previousPage");
		selenium.waitForPageToLoad("30000");
		verifyEquals(selenium.getTitle(), "Click Page 1");
		selenium.click("linkWithEnclosedImage");
		selenium.waitForPageToLoad("30000");
		verifyEquals(selenium.getTitle(), "Click Page Target");
		selenium.click("previousPage");
		selenium.waitForPageToLoad("30000");
		selenium.click("enclosedImage");
		selenium.waitForPageToLoad("30000");
		verifyEquals(selenium.getTitle(), "Click Page Target");
		selenium.click("previousPage");
		selenium.waitForPageToLoad("30000");
		selenium.click("extraEnclosedImage");
		selenium.waitForPageToLoad("30000");
		verifyEquals(selenium.getTitle(), "Click Page Target");
		selenium.click("previousPage");
		selenium.waitForPageToLoad("30000");
		selenium.click("linkToAnchorOnThisPage");
		verifyEquals(selenium.getTitle(), "Click Page 1");
		try { selenium.waitForPageToLoad("500"); fail("expected failure"); } catch (Throwable e) {}
		selenium.setTimeout("30000");
		selenium.click("linkWithOnclickReturnsFalse");
		Thread.sleep(300);
		verifyEquals(selenium.getTitle(), "Click Page 1");
		selenium.setTimeout("5000");
		selenium.open("../tests/html/test_click_page1.html");
		selenium.doubleClick("doubleClickable");
		assertEquals(selenium.getAlert(), "double clicked!");
	}
}

File: dotnet/test/webdriverbackedselenium.tests/TestClick.cs
using System;
using NUnit.Framework;

namespace Selenium.Tests
{
    [TestFixture]
    public class TestClick : SeleniumTestCaseBase
    {
        [Test]
        public void ShouldBeAbleToClick()
        {
            selenium.Open("/html/test_click_page1.html");
            Assert.AreEqual("Click here for next page", selenium.GetText("link"));
            selenium.Click("link");
            selenium.WaitForPageToLoad("30000");
            Assert.AreEqual("Click Page Target", selenium.GetTitle());
            selenium.Click("previousPage");
            selenium.WaitForPageToLoad("30000");
            Assert.AreEqual("Click Page 1", selenium.GetTitle());
            selenium.Click("linkWithEnclosedImage");
            selenium.WaitForPageToLoad("30000");
            Assert.AreEqual("Click Page Target", selenium.GetTitle());
            selenium.Click("previousPage");
            selenium.WaitForPageToLoad("30000");
            selenium.Click("enclosedImage");
            selenium.WaitForPageToLoad("30000");
            Assert.AreEqual("Click Page Target", selenium.GetTitle());
            selenium.Click("previousPage");
            selenium.WaitForPageToLoad("30000");
            selenium.Click("extraEnclosedImage");
            selenium.WaitForPageToLoad("30000");
            Assert.AreEqual("Click Page Target", selenium.GetTitle());
            selenium.Click("previousPage");
            selenium.WaitForPageToLoad("30000");
            selenium.Click("linkToAnchorOnThisPage");
            Assert.AreEqual("Click Page 1", selenium.GetTitle());
            try 
            { 
                selenium.WaitForPageToLoad("500");
                Assert.Fail("expected failure"); 
            }
            catch (Exception)
            {
            }
            selenium.SetTimeout("30000");
            selenium.Click("linkWithOnclickReturnsFalse");
            System.Threading.Thread.Sleep(300);
            Assert.AreEqual("Click Page 1", selenium.GetTitle());
            selenium.SetTimeout("5000");
            selenium.Open("/html/test_click_page1.html");
            //TODO: revisit when advanced user interactions is completed.
            //selenium.DoubleClick("doubleClickable");
            //Assert.AreEqual("double clicked!", selenium.GetAlert());
        }
    }
}
